saleor/product/utils.py

```diff
@@ -44,40 +46,63 @@ def products_with_availability(products, discounts, local_currency):
         'price_range_local_currency', 'discount_local_currency'))
 
 
-def get_availability(product, discounts=None, local_currency=None):
-    # In default currency
-    price_range = product.get_price_range(discounts=discounts)
-    undiscounted = product.get_price_range()
-    if undiscounted.min_price > price_range.min_price:
-        discount = undiscounted.min_price - price_range.min_price
+def calculate_discount(price, price_undiscounted):
+    if price_undiscounted.min_price > price.min_price:
+        return price_undiscounted.min_price - price.min_price
     else:
-        discount = None
+        return None
+
+
+def get_local_price_and_discount(price_range, price_range_undiscounted,
+                                 local_currency):
+    price_range_local = to_local_currency(price_range, local_currency)
+    undiscounted_local = to_local_currency(price_range_undiscounted,
+                                           local_currency)
+    if (undiscounted_local and
+            undiscounted_local.min_price > price_range_local.min_price):
+        discount_local_currency = (
+            undiscounted_local.min_price - price_range_local.min_price)
+    else:
+        discount_local_currency = None
 
-    # Local currency
+    return price_range_local, discount_local_currency
+
+
+def get_availability(is_available, price_range, price_range_undiscounted,
```

Comment:
Probably this method should become a private API part, because it doesn't have sense without a parent method like `get_product_availability`